packages/x-components/src/components/modals/x-modal.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <BaseEventsModal
+    class="x-modal"
+    :eventsToOpenModal="openEvents"
+    :eventsToCloseModal="closeEvents"
+    :bodyClickEvent="outOfModalClickEvent"
+    :animation="animation"
+  >
+    <slot />
+  </BaseEventsModal>
+</template>
+
+<script lang="ts">
+  import Component from 'vue-class-component';
+  import Vue from 'vue';
+  import { Prop } from 'vue-property-decorator';
+  import { XEvent } from '../../wiring/events.types';
+  import BaseEventsModal from './base-events-modal.vue';
+
+  /**
+   * A specialised version of a modal component, made to contain a full search application.
+   *
+   * @public
+   */
+  @Component({
+    components: {
+      BaseEventsModal
+    }
+  })
+  export default class XModal extends Vue {
```

Comment:
And about the Events names, we also should rename them, but let's wait until we have the modals Xmodule, because we have to rethink also the repeated Events for the same (Open/Close main modal). At the moment I would only rename the XModal

Reply:
Also I would rename the `OpenXModal` and `CloseXModal` components